### DISCUSSION TITLE
SQLite3 support for local development?

### DISCUSSION THREAD

**Discussion by todd-a-jacobs** [luckyframework/lucky#1704] [Feature requests & Ideas] [answered]
I'm extremely new to Crystal and to Lucky, although I have a strong background in Ruby on Rails. I'm used to being able to use SQLite3 for local development, and more robust databases in other environments.

I can see [from the documentation](https://luckyframework.org/guides/database/database-setup) that Lucky supports multiple databases, but I don't see an obvious way to either initialize a new Lucky project with SQLite3, or to modify it after initialization for an environment-specific database. This is possibly just a learning curve issue, but currently adds the overhead of installing and configuring PostgreSQL just to test out Lucky.

As something I consider a common use case, I'd like to request that wizard or initialization support for SQLite3 in development be supported "out of the box."

**Comment by jwoertink** [comment, marked as answer]
Hey, thanks for popping in! Lucky actually does not support multiple databases. Our ORM [Avram](https://github.com/luckyframework/avram/) only supports using Postgres.We've actually found that there's very minimal overhead to setting this up, and in the end, it actually works much better since it enables you to test a production setup locally which is something that's a lot more difficult to do in Rails. 

You can use SQLite, however, that setup becomes up to you, and with the latest release (0.30.0), it could prove to be more trouble than just using postgers from the start. I do have a plan to make using alternate ORMs easier in the next release, but it's a long slow process to get there.

As for setting up postgres, if you're on MacOS [Postgres app](https://postgresapp.com/) is a quick and easy install. Or, if you prefer using Docker, Lucky ships with a full container setup so you can just boot that and have everything ready to go. Just a `docker-compose up` :+1:  

Also feel free to drop in to our chat if you have more questions! https://luckyframework.org/chat

**Reply by todd-a-jacobs** [reply]
Thanks for the reply. I didn't realize it came with its own PostgreSQL container. I was trying to get the official Docker container for PostgreSQL to work with Lucky's setup, and was running into various issues, which is why I brought up the question. Now that I know it's got out-of-the-box support for a containerized DB, it's definitely less of a concern. If you want to mark this as resolved you can, unless you think there's something meritorious that might apply to future releases.
